pages/posts/light-clients.md

```diff
@@ -0,0 +1,260 @@
+---
+title: State of light clients, and how Portal can help
+date: 2023/2/01
+description: Explaining why is it important to have multiple client implementations.
+tag: client diversity, multiclient, Ethereum, archive
+author: jmc
+
+---
+
+## Key Takeaways
+
+- Light clients give access to Ethereum data to small devices, making it possible for apps, wallets, mobiles, bridges and IoT devices to interact with Ethereum. The associated trust assumptions depend upon the specific light client, ranging from completely trusting some RPC provider through to verifying both consensus and execution using SNARKs.
+- Light consensus clients are currently easier to build than light execution clients because a dedicated light client protocol for consensus clients was launched at The Merge.
+- The Portal Network enables users to get Beacon headers and `eth_getProof` responses from a peer-to-peer network, solving the problem of centralized third party RPC providers.
+- The Portal Network also provides a decentralized storage network for historical data, enabling EIP 4444 (history expiry). This allows for lighter full nodes.
+- Full nodes can be made lighter by expiring state data, but this requires multiple difficult protocol upgrades. (non inclusive list is Verkle tries, and address space expansion)
+- Proof systems such as SNARKs are already being used to verify sync committees, but they will likely be used to verify execution, such as the state transition function.
+- Further into the future, stateless clients will provide very lightweight access to Ethereum, but there are substantial prerequisites that must be shipped first, including Verkle tries.
```

Comment:
I don't think this is accurate.  Stateless clients are actually likely to not provide much of any access to Ethereum data since they will not have a state database...

Reply:
yea, thanks. This was a clumsy sentence. IIUC stateless clients will be useful because they can verify witnesses and that's enouhg for a validator to run on small hardware, and they might be able to participate in transaction gossiping. Is that about right?